src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -4245,6 +4245,46 @@ TEST_CASE("SimpleMultiheadAttentionTest", "[AttentionTest]")
  */
 TEST_CASE("JacobianMultiheadAttentionTest", "[AttentionTest]")
 {
+  // Check when query = key = value.
+  for (size_t i = 0; i < 5; ++i)
+  {
+    const size_t tgtSeqLen = 2;
+    const size_t embedDim = 4;
+    const size_t nHeads = 2;
+    const size_t batchSize = 1;
+
+    arma::mat input = arma::randu(embedDim * tgtSeqLen, batchSize);
+
+    MultiheadAttention<> module(embedDim, nHeads);
+    module.Key() = input;
+    module.Value() = input;
```

Comment:
I've set `query = key = value`, and it works (passed locally) without changing the Backward implementation.